### ISSUE TITLE
Initialising provider with generated SDK fails with `ProviderResource.__init__() got an unexpected keyword argument 'package_ref'`

### ISSUE THREAD

**Issue by VenelinMartinov** [pulumi/pulumi#17431]
### What happened?

Generated a local SDK for the OVH cloud provider with:
1. `pulumi new python --yes`
2. `pulumi package add terraform-provider registry.terraform.io/ovh/ovh`
3. echo sdks/ovh >> requirements.txt
4. `pulumi install`
5. `pulumi preview`

with the following program and got an error:

```
View in Browser (Ctrl+O): https://app.pulumi.com/pulumi/tf_prov_38/dev/previews/e2c5a48e-9228-4754-9a2d-ce4927054b6c

Loading policy packs...

     Type                 Name            Plan       Info
 +   pulumi:pulumi:Stack  tf_prov_38-dev  create     1 error; 3 messages

Policies:
    ✅ pulumi-internal-policies@v0.0.6

Diagnostics:
  pulumi:pulumi:Stack (tf_prov_38-dev):
    /Users/vvm/code/programs/tf_prov_38/venv/lib/python3.12/site-packages/pulumi_ovh/provider.py:218: RuntimeWarning: coroutine 'get_package' was never awaited
      super(Provider, __self__).__init__(
    RuntimeWarning: Enable tracemalloc to get the object allocation traceback

    error: Program failed with an unhandled exception:
    Traceback (most recent call last):
      File "/Users/vvm/code/programs/tf_prov_38/__main__.py", line 3, in <module>
        ovh.Provider("ovh")
      File "/Users/vvm/code/programs/tf_prov_38/venv/lib/python3.12/site-packages/pulumi_ovh/provider.py", line 190, in __init__
        __self__._internal_init(resource_name, *args, **kwargs)
      File "/Users/vvm/code/programs/tf_prov_38/venv/lib/python3.12/site-packages/pulumi_ovh/provider.py", line 218, in _internal_init
        super(Provider, __self__).__init__(
    TypeError: ProviderResource.__init__() got an unexpected keyword argument 'package_ref'


```

### Example

```python
import pulumi_ovh as ovh

ovh.Provider("ovh")
```

### Output of `pulumi about`

```
CLI
Version      3.134.1
Go Version   go1.23.1
Go Compiler  gc

Plugins
KIND      NAME                VERSION
language  python              3.134.1
resource  terraform-provider  0.2.0

Host
OS       darwin
Version  14.1.1
Arch     arm64

This project is written in python: executable='/Users/vvm/code/programs/tf_prov_38/venv/bin/python' version='3.12.0'

Current Stack: pulumi/tf_prov_38/dev

Found no resources associated with dev

Found no pending operations associated with dev

Backend
Name           pulumi.com
URL            https://app.pulumi.com/venelin-pulumi-corp
User           venelin-pulumi-corp
Organizations  venelin-pulumi-corp, demo, pulumi
Token type     personal

Dependencies:
NAME        VERSION
pip         24.2
pulumi_ovh  0.50.0
setuptools  75.1.0
wheel       0.44.0

Pulumi locates its logs in /var/folders/82/nqnqw81s1h56l5nv940f9mq00000gn/T/ by default
```

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by Frassle** [comment]
Not a regression, seems to be an issue with provider resources never got package_ref added to them in the core SDK. Looks like this never worked, should be an easy fix but will need a release.

**Comment by lunaris** [comment]
I think this will have been fixed by #17432
